backend/director/agents/web_search_agent.py

```
@@ -0,0 +1,189 @@
+                if knowledge_graph:
+                    title = knowledge_graph.get("source", {}).get("name")
+                    link = knowledge_graph.get("source", {}).get("link")
+                    description = knowledge_graph.get("description", "No description available.")
+                    if title and link:
+                        documents.append({
+                            "index": len(documents),
+                            "title": title,
+                            "link": link,
+                            "snippet": description
+                        })
+                        reranker_documents.append(f"Search Title: {result.get('title')}. Search Snippet: {result.get('snippet')}.")
```

Ruff (0.8.2)

Undefined name `result`

(F821)